lib/LedgerSMB/Magic.pm

```diff
@@ -0,0 +1,389 @@
+    HTTP_BAD_REQUEST
+    HTTP_FOUND
+    HTTP_INTERNAL_SERVER_ERROR
+    HTTP_OK
+    HTTP_SEE_OTHER
+    HTTP_UNAUTHORIZED
+
+    JRNL_GJ
+    JRNL_AR
+    JRNL_AP
+    JRNL_CR
+    JRNL_CD
+
+    MAX_DAYS_IN_MONTH
+    MAX_POST_SIZE_DEFAULT
+    MIN_PER_HOUR
+    MONEY_EPSILON
+    MONTHS_PER_QUARTER
+    MONTHS_PER_YEAR
+
+    NC_ENTITY
+    NC_INVOICE
+    NC_ENTITY_CREDIT_ACCOUNT
+    NC_JOURNAL_ENTRY
+
+    OEC_SALES_ORDER
+    OEC_PURCHASE_ORDER
+    OEC_QUOTATION
+    OEC_RFQ
+
+    PERL_TIME_EPOCH
+    RATIO_TO_PERCENT
+
+    RC_DEPRECIATION
+    RC_DISPOSAL
+    RC_IMPORT
+    RC_PARTIAL_DISPOSAL
+
+    SEC_PER_HOUR
+    SATURDAY
+    SUNDAY
+
+    YEARS_PER_CENTURY
+);
+
+use constant {
+
+    # temporal
+    CENTURY_START_YEAR => 2000,    # start of current century
+    FUTURE_YEARS_LIMIT => 20,      # go to last century if too far in future
+    SEC_PER_HOUR       => 3600,
+    MIN_PER_HOUR       => 60,
+    MONTHS_PER_QUARTER => 3,
+    MONTHS_PER_YEAR    => 12,
+    MAX_DAYS_IN_MONTH  => 31,
+    PERL_TIME_EPOCH    => 1900,
+    YEARS_PER_CENTURY  => 100,
+    SUNDAY             => 0,
+    SATURDAY           => 6,
+
+    # miscellany
+    MAX_POST_SIZE_DEFAULT      => 1024 * 1024 * 4,
+    MONEY_EPSILON => 0.001,         # XXX GAP/IFRS require .0001  maybe???
+                                    # I read that somewhere, maybe Celko --rir
+    RATIO_TO_PERCENT => 100,
+
+    # our batch classes
+    BC_AP               => 1,
+    BC_AR               => 2,
+    BC_PAYMENT          => 3,
+    BC_PAYMENT_REVERSAL => 4,
+    BC_GL               => 5,
+    BC_RECEIPT          => 6,
+    BC_RECEIPT_REVERSAL => 7,
+    BC_SALES_INVOICE    => 8,
+    BC_VENDOR_INVOICE   => 9,
+
+    # our entity classes
+    EC_VENDOR    => 1,
+    EC_CUSTOMER  => 2,
+    EC_EMPLOYEE  => 3,
+    EC_CONTACT   => 4,
+    EC_LEAD      => 5,
+    EC_REFERRAL  => 6,
+    EC_HOT_LEAD  => 7,
+    EC_COLD_LEAD => 8,
+
+    # our file classes:  files attached to sql records
+    FC_TRANSACTION => 1,
+    FC_ORDER       => 2,
+    FC_PART        => 3,
+    FC_ENTITY      => 4,
+    FC_ECA         => 5,
+    FC_INTERNAL    => 6,
+    FC_INCOMING    => 7,
+
+    # journal_type
+    JRNL_GJ => 1,
+    JRNL_AR => 2,
+    JRNL_AP => 3,
+    JRNL_CR => 4,
+    JRNL_CD => 5,
+
+    # our note classes
+    NC_ENTITY                => 1,
+    NC_INVOICE               => 2,
+    NC_ENTITY_CREDIT_ACCOUNT => 3,
+    NC_JOURNAL_ENTRY         => 5,
+
+    # our order entry classes
+    OEC_SALES_ORDER    => 1,
+    OEC_PURCHASE_ORDER => 2,
+    OEC_QUOTATION      => 3,
+    OEC_RFQ            => 4,
+
+    # our asset report classes
+    RC_DEPRECIATION     => 1,
+    RC_DISPOSAL         => 2,
+    RC_IMPORT           => 3,
+    RC_PARTIAL_DISPOSAL => 4,
+
+    # codes
+
+    # EDI
+    EDI_PATHNAME_MAX => 180,    # default max length of EDI pathname
+                                # TODO possible fencepost error XXX
+
+    # HTTP  These can be taken from HTTP::Status
```

Comment:
I thought I had indicated my agreement and my desire to clean up Magic.pm later for efficiency in the task.  I'd like to do that at the end of dealing with magic.  If that delay  is burdensome, let me know.  I was thinking of a second PR for the second half of magic numbers and finishing with Magic.pm--works for me but does it work for you?

http://www.1edisource.com/Portals/0/docs/850V5010.pdf gives descriptions of the 16 indices  of EDI items.  These could be the basis for constant names if we cannot find an authoritative source.  Last I knew, getting actual specifications was expensive.